zipkin-cassandra-core/src/main/java/org/twitter/zipkin/storage/cassandra/Repository.java

```diff
@@ -70,6 +70,23 @@
     private final PreparedStatement selectTraceDurationTail;
     private final Map<String,String> metadata;
 
+    private final ThreadLocal<Set<String>> writtenNames = new ThreadLocal<Set<String>>() {
+                private long hour = TimeUnit.MILLISECONDS.toHours(System.currentTimeMillis());
```

Comment:
What about solving it with Yuri suggestion?

> In this OSS implementation I'd suggest making this TTL externally configurable.

Then in any setup where the size of the thread-local maps become an issue it is just to set the cache ttl to a smaller value.